### PR TITLE
Split ClockUtils to improve Java compatibility

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/services/events/NodeSchedulerService.kt [hunk 1]
```diff
@@ -48,6 +60,45 @@ class NodeSchedulerService(private val services: ServiceHubInternal,
...
+        /**
+         * Wait until the given [Future] is complete or the deadline is reached, with support for [MutableClock] implementations
+         * used in demos or testing.  This will substitute a Fiber compatible Future so the current
+         * [co.paralleluniverse.strands.Strand] is not blocked.
+         *
+         * @return true if the [Future] is complete, false if the deadline was reached.
+         */
+        @Suspendable
+        @VisibleForTesting
+        fun awaitWithDeadline(clock: Clock, deadline: Instant, future: Future<*> = SettableFuture.create<Any>()): Boolean {
```
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/services/events/NodeSchedulerService.kt]: can we use the fully qualified name of SettableFuture here, or import it as GuavaSettableFuture, to make the distinction from quasar SettableFuture clearer.

File: node/src/main/kotlin/net/corda/node/utilities/ClockUtils.kt [hunk 2]
```diff
@@ -1,115 +0,0 @@
...
-import java.util.concurrent.atomic.AtomicLong
-import com.google.common.util.concurrent.SettableFuture as GuavaSettableFuture
-
-/**
- * The classes and methods in this file allow the use of custom Clocks in demos, simulations and testing
- * that might not follow "real time" or "wall clock time".  i.e. they allow time to be fast forwarded.
- *
- * We should try to make the Clock used in our code injectable (for tests etc) and to use the extensions below
- * to wait in our code, rather than <code>Thread.sleep()</code> or <code>Object.wait()</code> etc.
- */
```
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/utilities/ClockUtils.kt]: was this comment block intentionally dropped?

File: node/src/main/kotlin/net/corda/node/internal/MutableClock.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package net.corda.node.internal
+
+import rx.Observable
+import rx.Subscriber
+import rx.subscriptions.Subscriptions
+import java.time.Clock
+import java.util.concurrent.CopyOnWriteArraySet
+import java.util.concurrent.atomic.AtomicLong
+import com.google.common.util.concurrent.SettableFuture as GuavaSettableFuture
```
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/internal/MutableClock.kt]: unused import, intellij hasn't noticed

**Comment by rnicoll** [comment]
Rebased and addressed comments

**Pull request merged**